src/resolve/resolved-dns-stream.c

```diff
@@ -183,6 +183,36 @@ static int dns_stream_identify(DnsStream *s) {
         return 0;
 }
 
+static ssize_t dns_stream_writev(DnsStream *s, struct iovec *iov, int iovcnt) {
+        ssize_t r;
+
+        assert(s);
+        assert(iov);
+
+        if (s->tfo_salen) {
+                struct msghdr hdr = {};
+
+                hdr.msg_iov = iov;
+                hdr.msg_iovlen = iovcnt;
+                hdr.msg_name = &s->tfo_address->sa;
+                hdr.msg_namelen = s->tfo_salen;
+                r = sendmsg(s->fd, &hdr, MSG_FASTOPEN);
+                if (r < 0) {
+                        if (errno == EOPNOTSUPP) {
+                                mfree(s->tfo_address);
+                                s->tfo_salen = 0;
+                                errno = EAGAIN;
+                                connect(s->fd, (struct sockaddr*) &s->tfo_address->sa, s->tfo_salen);
```

Comment:
hmm, no error checking?

Reply:
also, and more importantly, you pass a ptr to an address you just freed a few lines up...

Also, the cast should not be necessary, as the `->sa` field already should have the right type